spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClient.java

```diff
@@ -93,12 +110,18 @@ public String description() {
 		return instances;
 	}
 
-	private Map<String, String> getLabels(String serviceName) {
-		final Service service = client.services().withName(serviceName).get();
-		if (service != null) {
-			return service.getMetadata().getLabels();
+	// always returns a new map that contain all the entries of the original map
+	// but with the keys prefixed
+	private Map<String, String> getMapWithPrefixedKeys(Map<String, String> map, String prefix) {
+		// when the prefix is empty just return an map with the same entries
+		if (prefix.isEmpty()) {
```

Comment:
use `StringUtils.hasText(prefix)` (assuming prefix defaults to null: see other comments).